src/support_code_library_builder/world.ts

```diff
@@ -6,7 +6,14 @@ export interface IWorldOptions {
   parameters: any
 }
 
-export default class World {
+export interface IWorld {
+  readonly attach: ICreateAttachment
+  readonly log: ICreateLog
+  readonly parameters: any
+  [key: string]: any
```

Comment:
I guess this is for arbitrary properties?
So, no way to avoid the `any` here?

What about deprecating that in a next major release? If users want extra parameters, they would have to implement their own strongly typed custom worlds? Or something else, but with strong types?

Reply:
Good question!

I think it's reflective of how many people use it in plain JS - setting some properties on `this` without the overhead of writing a custom class (many of our own acceptance tests do this). Once you want to have some methods, logic etc on the word then you'll naturally want to use your own class (for the type completion if nothing else).

We could make it stricter but it'd be a breaking change and would seem to hurt the developer ergonomics a bit IMO.

Reply:
I understand the need as a JavaScript developer. And I am not sure that would break things for JavaScript.

In the TypeScript world, if we want to enforce use of types, it seems less non-sense.

And yes, that would be a major breaking change

Reply:
Would you be okay with merging this now and picking the possible breaking change up in another issue?

Reply:
Yes, it's ok

It was more kinda general though